crates/hdk_derive/src/util.rs

```diff
@@ -12,28 +15,33 @@ pub fn to_snake_case(name: Option<String>, v_ident: &syn::Ident) -> String {
 
 pub fn ignore_enum_data(fields: &Fields) -> proc_macro2::TokenStream {
     match fields {
-        syn::Fields::Named(_) => quote::quote! {{..}},
-        syn::Fields::Unit => quote::quote! {},
-        syn::Fields::Unnamed(_) => quote::quote! {(_)},
+        Fields::Named(_) => quote::quote! {{..}},
+        Fields::Unit => quote::quote! {},
+        Fields::Unnamed(_) => quote::quote! {(_)},
     }
 }
 
 pub fn get_unit_ident(attrs: &[syn::Attribute]) -> syn::Ident {
     attrs
         .iter()
         .find(|a| {
-            a.path
+            a.path()
                 .segments
                 .last()
                 .is_some_and(|s| s.ident == "unit_enum")
         })
         .and_then(|a| darling::util::parse_attribute_to_meta_list(a).ok())
-        .and_then(|syn::MetaList { path, nested, .. }| {
+        .and_then(|syn::MetaList { path, tokens, .. }| {
+            let Ok(nested) =
+                Punctuated::<NestedMeta, Token![,]>::parse_terminated.parse(tokens.into())
+            else {
+                return None;
+            };
```

Comment:
A.gain. MetaList now holds a more generic token stream instead of nested metas. So the token stream needs to be parsed.
Returning None on parse error since the `unwrap_or_else()` will catch it.